peer/upstream/setup.go

```diff
@@ -55,8 +38,10 @@ func New(privateKey []byte, publicKey []byte, timeout time.Duration) (*Upstream,
 
 	shutdown := make(chan struct{})
 	stopPollingSig := make(chan struct{})
+	upstreamStr := fmt.Sprintf("upstream@%d", n)
```

Comment:
`name` is better, right?

Reply:
i would like to have more specific denotion, name can be remote client name